### DISCUSSION TITLE
Router context losing types infer

### DISCUSSION THREAD

**Discussion by igortas** [TanStack/router#1328] [Q&A] [answered]
Hi,

After setting up properly the router, I'm trying to access the ```context``` values from the ```loader``` and II face issues getting proper intellisense for the context types I'm injecting, in my case ```queryClient```.

Vscode can't detect the types and returns it as ```(parameter) context: {}```
Happens on the latest version

```
     <RouterProvider
          router={router}
          context={{
            queryClient
          }}
        />
```

**Comment by schiller-manuel** [comment]
you have to use  https://tanstack.com/router/v1/docs/framework/react/api/router/createRootRouteWithContextFunction



if it still does not work, please provide a minimal complete example

**Comment by igortas** [comment]
Hi @schiller-manuel, nothing wrong or different I'm doing

```router.tsx```

```
import { CircularProgress } from '@mui/material';
import { routeTree } from '../../routeTree.gen';
import { ErrorComponent, createRouter } from '@tanstack/react-router';

export const router = createRouter({
  routeTree,
  defaultPendingComponent: () => <CircularProgress />,
  defaultErrorComponent: ({ error }: any) => <ErrorComponent error={error} />,
  defaultPreload: 'intent',
  // Since we're using React Query, we don't want loader calls to ever be stale
  // This will ensure that the loader is always called when the route is preloaded or visited
  defaultPreloadStaleTime: 0
});

declare module '@tanstack/react-router' {
  interface Register {
    router: typeof router;
  }
}
```

```Provider.tsx```

```
const queryClient = new QueryClient();

export default function QueryClientProvider() {
  return (
    <Suspense fallback={<CircularProgress />}>
      <RQProvider client={queryClient}>
        <RouterProvider
          router={router}
          context={{
            queryClient
          }}
        />
      </RQProvider>
    </Suspense>
  );
}
```

This is the ```_root.tsx```

```
export const Route = createRootRouteWithContext()({
  component: () => <div>hello</div>,
  loader: ({ context: { queryClient } }) => { }
});

```

Here is the error:

![image](https://github.com/TanStack/router/assets/16512083/1a5367cb-2864-4a5f-8caa-66bf84707464)

**Reply by schiller-manuel** [reply, marked as answer]
you did not specify the generic type argument

https://tanstack.com/router/v1/docs/framework/react/api/router/createRootRouteWithContextFunction#troutercontext-generic

**Reply by tutods** [reply]
I'm facing some issue.
But I have the generic.

How you solve it?
